craft_parts/ctl.py

```diff
@@ -60,16 +60,16 @@ def client(cmd: str, args: List[str]):
         feedback_fifo = os.environ["PARTS_FEEDBACK_FIFO"]
     except KeyError as err:
         raise RuntimeError(
-            "{!s} environment variable must be defined.\nNote that this "
-            "utility is designed for use only in part scriptlets.".format(err)
+            f"{err!s} environment variable must be defined.\nNote that this "
+            f"utility is designed for use only in part scriptlets."
         ) from err
 
     data = {"function": cmd, "args": args}
 
-    with open(call_fifo, "w") as fifo:
+    with open(call_fifo, "w", encoding="utf-8") as fifo:
```

Comment:
Agreed `locale.getpreferredencoding(False)` is the safest option.  I'm not clear on Windows encoding in other languages/locales.
Resolving this comment as a duplicate since the conversation continues in [this thread](https://github.com/canonical/craft-parts/pull/164/files/87282e6dcf2de1b8f6f3f22a3f60dd6fba04ff50#r795993554)

Reply:
From what I understand from the documentation setting it to `locale.getpreferredencoding(False)` is already the default, so my preference would be to not declare it explicitly and disable the warning instead.

Reply:
Disabled `unspecified-encoding` and reverted all file i/o to implicit default.